src/test/java/seedu/duke/atomichabit/AtomicHabitTest.java

```diff
@@ -41,6 +42,41 @@ public void addHabit_invalidCommand_exceptionThrown() {
         }, "The following is an invalid command:\n"
                 + command);
     }
+
+    @Test
+    public void updateHabit_correct_output() throws AtomicHabitException {
+        addHabit_correct_output();
+        String payload = "junit test";
+        String habitIndex = "1";
+        Command updateCommand = new UpdateCommand(habitIndex);
+        String expectedUpdateHabitOutput = "The following habit has been incremented! Keep up the good work!"
+                + System.lineSeparator()
+                + habitIndex + "." + payload + " " + "[2]"
+                + System.lineSeparator();
+        Assertions.assertEquals(expectedUpdateHabitOutput, updateCommand.execute(habitList).getCommandResult());
+    }
+
+    @Test
+    public void updateHabit_indexNotInteger_exceptionThrown() throws AtomicHabitException {
+        // Test false command by user
+        addHabit_correct_output();
+        String habitIndex = "a";
+        Command updateCommand = new UpdateCommand(habitIndex);
+        Assertions.assertThrows(AtomicHabitException.class, () -> {
+            updateCommand.execute(habitList);
+        });
+    }
+
+    @Test
+    public void updateHabit_indexOutOfBounds_exceptionThrown() throws AtomicHabitException {
+        // Test false command by user
+        addHabit_correct_output();
+        String habitIndex = "1000";
+        Command updateCommand = new UpdateCommand(habitIndex);
+        Assertions.assertThrows(AtomicHabitException.class, () -> {
+            updateCommand.execute(habitList);
+        });
+    }
```

Comment:
A nit, but for fuller coverage, test if negative values throw the appropriate error as well